docs/proposals/signal-api.md

```diff
@@ -0,0 +1,194 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Signal API
+
+This document describes signal API effort in Kubernetes.
+
+## Abstract
+
+The goal is to provide a possibility to send signals to container entrypoint application. We are going to introduce
+a concept of *notification* in the pod's spec, which will be a mapping of a user-defined name to exact POSIX signal
+name. In example, if user wants to send SIGHUP signal on the *bump* notification, then he/she will need to define
+that behavior in the PodSpec and then use the *bump* term when calling the notification API resource.
+
+Once me make agreement on the API part, the document will be extended with the implementation details.
+
+One thing that needs to be noticed is that restarting scenarios are not a goad of this proposal. The only goal here
+is to send single signals, no more, no less. More complicated restart scenarios (like sending SIGTERM, then waiting
+for graceful termination, then sending SIGKILL after reaching timeout) may need another resource in Kubernetes API,
+which will not expose any POSIX terminology to the user.
+
+#### Containers and pods
+
+Despite the fact that Docker supports shared PID namespaces, Kubernetes doesn't, which means that signals can be
```

Comment:
@thockin Shared PID ns wouldn't change anything in the current shape of this spec, I think. Thanks for pointing that, I will remove this paragraph.

@goblain Well, of course you could do this when having shared PID ns in k8s, but... why? We are thinking about using native kill features of container runtimes to avoid such hacks.